### PR TITLE
Translate-deep-copy

### PR DESCRIPTION
**Pull request by gregWDumont**
Update links except for `Object.create()`, "HTML DOM API", and `structuredClone()` which has not been translated yet.

I have not modified it but there is an issue with this as it doesn't appear on the EN-US repo: [`window`](/en-US/docs/Web /API/Fenêtre).

### CODE REVIEW

**Comment by gregWDumont** [comment]
These links are supposed to replace the ones from the EN-US/docs/ source. I can't figure out if I should leave the initial links or use the ones from the translated repo?!

**Comment by cw118** [comment]
Links to other MDN Web Docs pages should start with `/fr/docs/`. For example, to create a link to the (French) browser glossary page (`https://developer.mozilla.org/fr/docs/Glossary/Browser`), one writes: `[Navigateur](/fr/docs/Glossary/Browser)`. Internal links like these should point to `fr` even if the page does not yet exist in French :)

File: files/fr/glossary/deep_copy/index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+```js
+let ingredients_list = ["noodles",{"list":["eggs","flour","water"]}];
+let ingredients_list_deepcopy = JSON.parse(JSON.stringify(ingredients_list));
+
+// Modifiez la valeur de la propriété 'list' dans ingredients_list_deepcopy.
+ingredients_list_deepcopy[1].list = ["rice flour","water"]
+// La propriété 'list' ne change pas dans ingredients_list.
+console.log(ingredients_list[1].list);
+// Array(3) [ "eggs", "flour", "water" ]
```
Comment (cw118) [inline files/fr/glossary/deep_copy/index.md:26]:
```suggestion
let liste_ingredients = ["nouilles", {"liste": ["œufs", "farine", "eau"]}];
let liste_ingredients_copieprofonde = JSON.parse(JSON.stringify(liste_ingredients));

// Modifier la valeur de la propriété 'liste' dans liste_ingredients_copieprofonde
liste_ingredients_copieprofonde[1].liste = ["farine de riz", "water"];
// La propriété 'liste' ne change pas dans liste_ingredients
console.log(liste_ingredients[1].liste);
// Array(3) [ "œufs", "farine", "eau" ]
```

File: files/fr/glossary/deep_copy/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+Comme le montre le code ci-dessus, étant donné qu'une copie complète ne partage aucune référence avec son objet source, toute modification apportée à la copie complète n'affecte pas l'objet source.
+
+Cependant, alors que l'objet dans le code ci-dessus est assez simple pour être [sérialisable](/files/fr/Glossary/Serialization), de nombreux objets JavaScript ne sont pas du tout sérialisables - par exemple, les [fonctions](/files/fr/Web/JavaScript/Guide/Functions) (avec des fermetures), les [symboles](/files/fr/Web/JavaScript/Reference/Global_Objects/Symbol), les objets qui représentent des éléments HTML dans le [HTML DOM API](/en-US/docs/Web/API/HTML_DOM_API), les données récursives, et bien d'autres cas. L'appel de `JSON.stringify()` pour sérialiser les objets échouera dans ce cas. Il n'y a donc aucun moyen de faire des copies complètes de tels objets.
```
Comment (cw118) [inline files/fr/glossary/deep_copy/index.md:31]:
```suggestion
Cependant, alors que l'objet dans le code ci-dessus est assez simple pour être [sérialisable](/fr/docs/Glossary/Serialization), de nombreux objets JavaScript ne sont pas du tout sérialisables — par exemple, les [fonctions](/fr/docs/Web/JavaScript/Guide/Functions) (avec des fermetures), les [symboles](/fr/docs/Web/JavaScript/Reference/Global_Objects/Symbol), les objets qui représentent des éléments HTML dans [l'API HTML DOM](/fr/docs/Web/API/HTML_DOM_API), les données récursives, et bien d'autres cas. L'appel de `JSON.stringify()` pour sérialiser ces objets échouera. Il n'y a donc aucun moyen de faire des copies complètes de tels objets.
```

File: files/fr/glossary/deep_copy/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Pour les objets qui _sont_ sérialisables, vous pouvez également utiliser le [`structuredClone()`](/en-US/docs/Web/API/structuredClone) méthode pour créer des copies profondes. `structuredClone()` a l'avantage de permettre {{Glossary("transferable objects")}} dans la source pour être _transféré_ vers la nouvelle copie, plutôt que simplement cloné. Mais notez que `structuredClone()` n'est pas une fonctionnalité du langage JavaScript lui-même - c'est plutôt une fonctionnalité des navigateurs et de tout autre environnement d'exécution de JavaScript qui implémente un objet global comme [`window`](/en-US/docs/Web /API/Fenêtre). Et appeler `structuredClone()` pour cloner un objet non sérialisable échouera de la même manière que l'appel de `JSON.stringify()` pour le sérialiser échouera.
```
Comment (cw118) [inline files/fr/glossary/deep_copy/index.md:33]:
```suggestion
Pour les objets qui _sont_ sérialisables, vous pouvez également utiliser la méthode [`structuredClone()`](/fr/docs/Web/API/structuredClone) pour créer des copies profondes. `structuredClone()` est avantageuse car elle permet de _transférer_ des [objets transférables](/fr/docs/Glossary/Transferable_objects) dans la source vers la nouvelle copie, plutôt que de simplement les cloner. Mais notez que `structuredClone()` n'est pas une fonctionnalité du langage JavaScript lui-même — il s'agit plutôt d'une fonctionnalité des navigateurs et de tout autre environnement d'exécution de JavaScript qui implémente un objet global comme [`window`](/fr/docs/Web/API/Window), et appeler `structuredClone()` pour cloner un objet non sérialisable échouera de la même manière que l'appel de `JSON.stringify()` pour le sérialiser échouera.
```

File: files/fr/glossary/deep_copy/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Pour approfondir
```
Comment (cw118) [inline files/fr/glossary/deep_copy/index.md:35]:
```suggestion
## Voir aussi
```

**Review by cw118 — CHANGES_REQUESTED** [review]
Quelques suggestions :) cela dit, merci beaucoup @gregWDumont !

File: files/fr/glossary/deep_copy/index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+- [copie superficielle](/en-US/docs/Glossary/Shallow_copy)
+- [`window.structuredClone()`](/en-US/docs/Web/API/structuredClone)
```
Comment (cw118) [inline files/fr/glossary/deep_copy/index.md:38]:
```suggestion
- [Copie superficielle](/fr/docs/Glossary/Shallow_copy)
- [`window.structuredClone()`](/fr/docs/Web/API/structuredClone)
```

j'avais loupé cette partie :/

**Review by cw118 — CHANGES_REQUESTED** [review]

**Review by cw118 — APPROVED** [review]
Merci beaucoup pour cette traduction @gregWDumont ! 

Je me suis permis de pousser un commit rajoutant la clé `l10n sourceCommit` (voir https://github.com/orgs/mdn/discussions/129) :)

**Pull request merged**